tests/test_gvmd_scanreport_datatransformation.py

```diff
@@ -0,0 +1,170 @@
+        if with_optional
+        else None,
+        'threat': random.choice(threats) if with_optional else None,
+        'severity': '4.3' if with_optional else None,
+        'qod': gen_qod() if with_optional else None,
+        'description': _random_text(254) if with_optional else None,
+    }
+
+
+def gen_gmp() -> Dict:
+    return {'version': '21.04'}
+
+
+def gen_count(count: int = random.randint(1, 100)) -> Dict:
+    return {'count': count}
+
+
+def gen_identifiable() -> dict:
+    return {
+        'id': uuid.uuid1().hex,
+        'name': _random_text(10),
+        'comment': _random_text(150),
+    }
+
+
+def gen_task() -> Dict:
+    return {
+        **gen_identifiable(),
+        'target': {**gen_identifiable(), 'trash': _random_text(2),},
+        'progress': '100',
+    }
+
+
+def gen_filtered() -> Dict:
+    return {'full': _random_text(25), 'filtered': _random_text(12)}
+
+
+def gen_result_count() -> Dict:
+    return {
+        **gen_filtered(),
+        'debug': gen_filtered(),
+        'hole': gen_filtered(),
+        'info': gen_filtered(),
+        'log': gen_filtered(),
+        'warning': gen_filtered(),
+        'false_positive': gen_filtered(),
+        'text': _random_text(25),
+    }
+
+
+oids = [uuid.uuid1().hex for _ in range(5)]
+
+
+def gen_report(with_optional=False) -> Dict:
+    return {
+        'id': uuid.uuid1().hex if with_optional else None,
+        'gmp': gen_gmp() if with_optional else None,
+        'scan_run_status': '100' if with_optional else None,
+        'hosts': gen_count(1) if with_optional else None,
+        'closed_cves': gen_count() if with_optional else None,
+        'vulns': gen_count() if with_optional else None,
+        'os': gen_count() if with_optional else None,
+        'apps': gen_count() if with_optional else None,
+        'ssl_certs': gen_count() if with_optional else None,
+        'task': gen_task() if with_optional else None,
+        'timestamp': '2342' if with_optional else None,
+        'scan_start': '2020-07-01T21:00' if with_optional else None,
+        'timezone': 'timezone_abbrev' if with_optional else None,
+        'timezone_abbrev': 'UTC' if with_optional else None,
+        'scan_end': '2020-07-01T21:00' if with_optional else None,
+        'erros': gen_count() if with_optional else None,
+        'results': {
+            'max': '{}'.format(random.randint(1, 1000))
+            if with_optional
+            else None,
+            'start': '{}'.format(random.randint(0, 1000))
+            if with_optional
+            else None,
+            'result': [
+                gen_result(
+                    '{}'.format(i % 2), oids[i % len(oids)], with_optional
+                )
+                for i in range(250)
+            ],
+        },
+        'severity': gen_filtered() if with_optional else None,
+        'result_count': gen_result_count() if with_optional else None,
+    }
+
+
+@pytest.mark.parametrize("scan_result", [gen_report(True)])
```

Comment:
For boolean values it is often better to pass the argument as keyword because it improves the readablity.
```suggestion
@pytest.mark.parametrize("scan_result", [gen_report(with_optional=True)])
```